Added cpp solution for 238

shruthi019 opened the pull request:
### Your checklist for this pull request
🚨Please review the [guidelines for contributing](https://github.com/larissalages/code_problems/blob/master/CONTRIBUTING.md) to this repository.

- [x] Make sure your files are in the right folder!
- [x] Make sure you add the extension of the file (.cpp, .python, .java, etc).
- [x] If your code is a solution for a code problem, please tell which one.
- [x] Make sure your code is working, test for a lot of cases.

### Description
I added the solution in C++ for problem 238 (Product of Array Except Self) in LeetCode.

:heart: Thank you!

larissalages reviewed (changes requested):
Can you solve the conflicts with master? thanks!

shruthi019 commented:
Hey, I saw that a solution was already added for this problem, hence there was a conflict. I cleaned the code and made it more readable, and corrected a typo.
